cmake/inputs/patches/ep_gcssdk/build.patch

```diff
@@ -0,0 +1,339 @@
+From 2b9a750a5b00d3021aaceff1ac738a33b29c3da1 Mon Sep 17 00:00:00 2001
+From: Joe maley <joe@tiledb.io>
+Date: Wed, 11 Mar 2020 11:26:06 -0400
+Subject: GCS Build Patch for TileDB
+
+---
+ super/CMakeLists.txt                         |  9 ++-
+ super/external/c-ares.cmake                  |  2 +
+ super/external/crc32c.cmake                  |  2 +
+ super/external/curl.cmake                    | 64 --------------------
+ super/external/google-cloud-cpp-common.cmake |  2 +
+ super/external/googleapis.cmake              |  2 +
+ super/external/googletest.cmake              |  2 +
+ super/external/grpc.cmake                    |  5 +-
+ super/external/protobuf.cmake                |  4 +-
+ super/external/ssl.cmake                     | 26 --------
+ super/external/zlib.cmake                    | 47 --------------
+ 11 files changed, 21 insertions(+), 144 deletions(-)
+ delete mode 100644 super/external/curl.cmake
+ delete mode 100644 super/external/ssl.cmake
+ delete mode 100644 super/external/zlib.cmake
+
+diff --git a/super/CMakeLists.txt b/super/CMakeLists.txt
+index 9cc916aed..a5d2680e8 100644
+--- a/super/CMakeLists.txt
++++ b/super/CMakeLists.txt
+@@ -23,7 +23,6 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+ set(GOOGLE_CLOUD_CPP_ROOT "${PROJECT_SOURCE_DIR}/..")
+ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}")
+ 
+-include(external/curl)
+ include(external/crc32c)
+ include(external/grpc)
+ include(external/googleapis)
+@@ -35,7 +34,7 @@ set_external_project_build_parallel_level(PARALLEL)
+ set_external_project_vars()
+ 
+ set(GOOGLE_CLOUD_CPP_DEPENDENCIES_LIST
+-    grpc-project curl-project crc32c-project googleapis-project
++    grpc-project crc32c-project googleapis-project
+     googletest-project google-cloud-cpp-common-project)
+ 
+ include(ExternalProject)
+@@ -54,8 +53,12 @@ ExternalProject_Add(
+                -DCMAKE_PREFIX_PATH=${GOOGLE_CLOUD_CPP_PREFIX_PATH}
+                -DCMAKE_INSTALL_RPATH=${GOOGLE_CLOUD_CPP_INSTALL_RPATH}
+                -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
++               -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
++               -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
++               -DBUILD_SAMPLES=${BUILD_SAMPLES}
++               -DCMAKE_CXX_FLAGS=-fPIC
```

Comment:
This should be passed through from the top-level project, I think.

Reply:
I tried that at one point but it didn't work. Could have been a programming issue, but it could also be that the CXX_FLAGS are overwritten. It would probably work by defining some auxiliary flag like (AUX_MAKE_CXX_FLAGS). Either way, I'll create a ticket and return to this later. I don't wanna get hung up on the build for too long before starting on the programming.

Reply:
https://app.clubhouse.io/tiledb-inc/story/1710/gcs-sdk-nested-compile-flags

Reply:
Makes sense, SGTM!